### PR TITLE
[AB#89606] Fix to change the WorkitemKeySequence data type to BIGINT

### PR DESCRIPTION
**Pull request by shivakumar-ms**
## Description
Fix to change the WorkitemKeySequence data type to BIGINT.

## Related issues
[AB#89606](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/89606)

## Testing
successful unit and integration Test runs.

### CODE REVIEW

File: src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/17.diff.sql [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+IF EXISTS (SELECT * FROM sys.sequences WHERE Name = 'WorkitemKeySequence' AND system_type_id = 56)
+    BEGIN
+
+        DECLARE @sql NVARCHAR(MAX);
+        DECLARE @newWorkitemKeyStartVal INT;
+
+        -- Daily average is at ~50. Hence we buffer 100 on top of the last workitem key sequence
+        SET @newWorkitemKeyStartVal = (NEXT VALUE FOR dbo.WorkitemKeySequence) + 100;
+
+        SET @sql = CONCAT(N'DROP SEQUENCE dbo.WorkitemKeySequence; CREATE SEQUENCE dbo.WorkitemKeySequence AS BIGINT START WITH ', STR(@newWorkitemKeyStartVal), 'INCREMENT BY 1 MINVALUE ' + STR(@newWorkitemKeyStartVal) + ' NO CYCLE CACHE 10000');
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/17.diff.sql:24]: Is the dynamic sql execution will also be inside the transaction? Or should we have a separate transaction inside the dynamic sql. What is the drop succeeded and the create failed?

File: src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/17.diff.sql [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        EXEC sys.sp_executesql @sql;
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/17.diff.sql:25]: Do we know that we have enough permission to execute or should we have to GRANT permission?

File: src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/17.diff.sql [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ - IMPORTANT: Avoid rebuiling indexes inside the transaction, it locks the table during the transaction.
+******************************************************************************************/
+
+SET XACT_ABORT ON
+
+BEGIN TRANSACTION
+GO
+
+-- Data Type INT = 56
+IF EXISTS (SELECT * FROM sys.sequences WHERE Name = 'WorkitemKeySequence' AND system_type_id = 56)
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/17.diff.sql:15]: Did you had a chance to execute? I get "The COMMIT TRANSACTION request has no corresponding BEGIN TRANSACTION." error

File: src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/17.diff.sql (hunk 1 above)
Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/17.diff.sql:24]: outer transaction should still take care of this scenario. if needed, we can add an inner transaction part of dynamic sql.

File: src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/17.diff.sql (hunk 2 above)
Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/17.diff.sql:25]: "sp_executesql" requires the user to be a public member. Permission is more important for the script it is executing. In this case, DROP/CREATE a sequence.

File: src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/17.diff.sql (hunk 3 above)
Comment (shivakumar-ms) [reply src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/17.diff.sql:15]: I ran it through in my local. and also through the schema upgrade tests. I did not run into the transaction error. Will nuke my database and try it again.

**Review by bcarthic — APPROVED** [review]

**Review by alishahahmed — APPROVED** [review]

**Pull request merged**